File: shared/script-types/examples/bugdom2-main.ts
import type { Bugdom2LifecycleModule } from "../src/games/bugdom2";
import { hasTag, makeVerticalBobOffset } from "./helpers";

const PLAYER_TAG = "bugdom2.player";
const PLAYER_BOB_SPEED = 5;
const PLAYER_BOB_HEIGHT = 18;

export function isBugdom2Player(tags: readonly string[]): boolean {
  return hasTag(tags, PLAYER_TAG);
}

export const onObjectFrame: NonNullable<
  Bugdom2LifecycleModule["onObjectFrame"]
> = (ctx) => {
  if (!isBugdom2Player(ctx.tags)) {
    return;
  }

  return makeVerticalBobOffset(
    ctx.levelTimeSeconds,
    PLAYER_BOB_SPEED,
    PLAYER_BOB_HEIGHT,
  );
};

File: shared/script/examples/bugdom2-main.lua
local helpers = require("helpers")
local module = {}

local PLAYER_TAG = "bugdom2.player"
local PLAYER_BOB_SPEED = 5
local PLAYER_BOB_HEIGHT = 18

function module.onObjectFrame(ctx)
    if not helpers.hasTag(ctx.tags, PLAYER_TAG) then
        return nil
    end

    return helpers.makeVerticalBobOffset(
        ctx.levelTimeSeconds,
        PLAYER_BOB_SPEED,
        PLAYER_BOB_HEIGHT
    )
end

return module
